libyelp/yelp-settings.c

```diff
@@ -599,11 +599,18 @@ yelp_settings_get_default (void)
     static GMutex mutex;
     static YelpSettings *settings = NULL;
     g_mutex_lock (&mutex);
-    if (settings == NULL)
-	settings = g_object_new (YELP_TYPE_SETTINGS,
-				 "gtk-settings", gtk_settings_get_default (),
-				 "gtk-icon-theme", gtk_icon_theme_get_default (),
-				 NULL);
+    if (settings == NULL) {
+        const gchar * yelp_skip_gui = g_getenv("YELP_SKIP_GUI");
+        if (yelp_skip_gui) {
+            settings = g_object_new (YELP_TYPE_SETTINGS, NULL);
+        }
+        else {
+            settings = g_object_new (YELP_TYPE_SETTINGS,
+                                     "gtk-settings", gtk_settings_get_default (),
+                                     "gtk-icon-theme", gtk_icon_theme_get_default (),
+                                     NULL);
```

Comment:
We want to skip the GUI part, when coming from the yelp search provider.
I already mentioned this in the commit [msg](https://github.com/endlessm/yelp/pull/40/commits/6805979ef07167873dafadcde70e8fb82721f106), maybe I should add a comment in the code as well?